src/safeds/exceptions/_generic.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+
+class OutOfBoundsError(ValueError):
+    """A generic exception that can be used to signal that a (float) value is outside its expected range."""
+
+    def __init__(self, actual: float, *, lower_bound: Bound | None = None, upper_bound: Bound | None = None):
+        """
+        Initialize an OutOfBoundsError.
+
+        Parameters
+        ----------
+        actual: float
```

Comment:
Should we treat passing `nan` as `actual` as another (value) error?

Reply:
This check could also be added to `Bound.__init__`.

Reply:
I can't think of a situation where this would make sense, so I added a check in `Bound.__init__`, raising a `ValueError` if `nan` is given.
If the need to pass `nan` ever occurs, we can allow it then.

Reply:
Makes sense. Then we should also check `actual` in `OutOfBoundsError.__init__`.

Reply:
Added checks for that.